### PR TITLE
Ensure rolling raises for incorrect inputs

### LINKED ISSUE #490: Ensure that rolling raises for incorrect inputs
float windows, float min periods, negative windows, ... see test_rolling_raises for reference in dask/dask

this is probably surfaced through pandas

### PR DESCRIPTION
**Pull request by milesgranger**
Close #490

### CODE REVIEW

**Comment by milesgranger** [comment]
Removed the test_rolling_raises cases w/ `axis` keyword arg as pandas is raising, let me know if we still want to keep those for now though! 

> FutureWarning: The 'axis' keyword in DataFrame.rolling is deprecated and will be removed in a future version. Call the method without the axis keyword instead.

File: dask_expr/_rolling.py [hunk 1]
```diff
@@ -221,10 +221,20 @@ class Rolling:
...
+    def __init__(
+        self,
+        obj,
+        window,
+        groupby_kwargs=None,
+        groupby_slice=None,
+        min_periods=None,
+        center=False,
+        win_type=None,
+        axis=0,
```
Comment (phofl) [inline dask_expr/_rolling.py:236]: let's remove this keyword completely

File: dask_expr/_rolling.py [hunk 2]
```diff
@@ -234,9 +244,28 @@ def __init__(self, obj, window, groupby_kwargs=None, groupby_slice=None, **kwarg
...
+
+        # Allow pandas to raise if appropriate
+        obj._meta.rolling(window, **self.kwargs)
+
+        # Using .rolling(window='2s'), pandas will convert the
+        # offset str to a window in nanoseconds. But pandas doesn't
+        # accept the integer window with win_type='freq', so we store
+        # that information here.
+        # See https://github.com/pandas-dev/pandas/issues/15969
+        self._win_type = None if isinstance(self.window, int) else "freq"
```
Comment (phofl) [inline dask_expr/_rolling.py:262]: why is this needed?

File: dask_expr/_rolling.py (hunk 1 above)
Comment (milesgranger) [reply dask_expr/_rolling.py]: Done and done https://github.com/dask-contrib/dask-expr/pull/497/commits/cbc057b0c961738e75f3f2d789f84bcab5268751

**Comment by phofl** [comment]
thx
